packages/hub/src/components/NavigationBlocker/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import { use, useCallback, useEffect, useId } from "react";
+
+import { NavigationBlockerContext } from "./context";
+
+// Be very careful with this hook! `blocker` parameter, if set, must have stable identity.
+// If it's an inline function, it will cause an infinite render loop:
+// - any change in blocker will cause all consumers of this hook to re-render
+// - this will cause the blocker to re-create, causing another re-render
+export function useBlockNavigation(blocker: () => boolean = () => true) {
+  const { dispatch } = use(NavigationBlockerContext);
+
+  const key = useId();
+
+  useEffect(() => {
+    dispatch({
+      type: "addBlocker",
+      payload: {
+        key,
+        blocker,
```

Comment:
Renamed to `check`. I agree that "blocker" is problematic because you can mistake it for the function that _does_ the blocking.

On `check` vs `shouldBlockFn`, because we pass the callback around, I think of it as an "object" (not just verb/action), so I feel like `check` hits a nice sweet spot in that regard.